### ISSUE TITLE
`no-dupe-characters-character-class` behaves strangely with `\s`

### ISSUE THREAD

**Issue by RunDevelopment** [ota-meshi/eslint-plugin-regexp#87]
For the character class `[\xA0-\uFFFF\s]` (all spaces and non-ascii), I get the following error message:

```
Unexpected intersection of "\xA0-\uFFFF" and "\s" was found "\u00a0".
```

I find it a little strange that this gets reported as an error. Is this correct?

Assuming that it is, I can just skip `\xA0` easily, so let's re-run the rule on `/[\xA1-\uFFFF\s]/`. 

> No Errors.

Now that is strange. `\xA1-\uFFFF` and `\s` aren't disjoint. They both share [quite a few characters](https://github.com/RunDevelopment/refa/blob/2bb82ef17418cd9c27ec16c55e1ce792e103528f/src/js/util.ts#L53-L59). Why doesn't `\u1680` get reported as well?

**Comment by ota-meshi** [comment]
Thank you for posting this issue!

It is my intended behavior that `\u1680` is not reported. Duplicate checking for escape character classes and character ranges is only reported when the character range does not need to be split.
But, it is a bug that it is not reported if the character range contains all the characters of the escape character class.

Do you think you should also report if need to split the character range? If so I can add options.

**Comment by RunDevelopment** [comment]
> Duplicate checking for escape character classes and character ranges is only reported when the character range does not need to be split.

I think this is a good strategy. It only produces this strange result because `\s` contains a lot of characters. Maybe it would be a good idea to ignore overlap with non-trivial character sets (e.g. `\s`, `\p{L}`)?

But only overlap. If another element is a subset of a non-trivial character set, then this should be reported (e.g. `[\t\s]`).

**Comment by ota-meshi** [comment]
I changed the PR #148 to report if the escape character class is completely included in the escape character class. Cases of these partial overlaps will not be reported. e.g. `/[\w\p{L}]/u`
